src/robot/running/handlers.py

```diff
@@ -382,3 +382,26 @@ def _parse_arguments(self, handler_method):
         parser = JavaArgumentParser(type='Test Library')
         signatures = self._get_signatures(handler_method)
         return parser.parse(signatures, self.library.name)
+
+
+class EmbeddedArgsTemplate(object):
+
+    def __init__(self, embedded, orig_handler):
+        self.orig_handler = orig_handler
+        self._orig_run = orig_handler._run
+        self.orig_handler._run = self._run
```

Comment:
I think not supporting that usage is fine for now, I'll be happy to help refactor if popular demand for that use case requires it.